_examples/signals-to-json.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+/* Example usage, showing systemd unit property changes:
+ *
+ *   signals-to-json -systemBus=true -filter="type='signal',interface='org.freedesktop.DBus.Properties',member='PropertiesChanged',arg0='org.freedesktop.systemd1.Unit',path_namespace='/org/freedesktop/systemd1/unit'"
+ *
+ */
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/charles-dyfis-net/go-dbus"
+	"os"
+)
+
+var cfg struct {
+	Filter string
+	SystemBus bool
+}
+
+func main() {
+	var conn *dbus.Conn
+	var err error
+
+	flag.BoolVar(&cfg.SystemBus, "systemBus", false, "Use system rather than session bus")
+	flag.StringVar(&cfg.Filter, "filter", "", "Filter to apply to select signals (default looks for systemd service property changes)")
+	flag.Parse()
+
+	if(flag.NArg() != 0) {
+		fmt.Fprintf(os.Stderr, "Unrecognized argument seen\n")
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	if(cfg.SystemBus) {
+		conn, err = dbus.SystemBus()
+	} else {
+		conn, err = dbus.SessionBus()
+	}
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Failed to connect to system bus:", err)
```

Comment:
Since you're using flags above to use the session or system bus, this error message may be misleading.